pkg/sink/elasticsearch.go

```diff
@@ -154,13 +162,19 @@ func (e *Elasticsearch) flushIndex(ctx context.Context, indexCfg config.ELSIndex
 			},
 		}
 		_, err := e.client.CreateIndex(indexName).BodyJson(mapping).Do(ctx)
-		if err != nil {
+		if err != nil && err.(*elastic.Error).Details.Type != "resource_already_exists_exception" {
 			return fmt.Errorf("while creating index: %w", err)
 		}
```

Comment:
```suggestion
		if err != nil && elastic.ErrorReason(err) != "resource_already_exists_exception" {
			return fmt.Errorf("while creating index: %w", err)
		}
```
BTW please move this reason to consts 👍